server/api/controllers/user.js

```diff
@@ -388,13 +388,15 @@ const requestResetPassword = (req, res) => {
 		domain
 	);
 
-	if (typeof email !== 'string' || !isEmail(email)) {
+	const emailPattern = /^[a-zA-Z0-9_\-.]+@[a-zA-Z0-9\-]+\.[a-zA-Z0-9\-.]+$/;
```

Comment:
The isEmail library is doing the same thing. It doesnt seem necessary to do regex here.

Reply:
I checked isEmail function, its allow some special characters, e.g: !, #, %, &, ...

`var emailUserPart = /^[a-z\d!#\$%&'\*\+\-\/=\?\^_`{\|}~]+$/i;
var quotedEmailUser = /^([\s\x01-\x08\x0b\x0c\x0e-\x1f\x7f\x21\x23-\x5b\x5d-\x7e]|(\\[\x01-\x09\x0b\x0c\x0d-\x7f]))*$/i;
var emailUserUtf8Part = /^[a-z\d!#\$%&'\*\+\-\/=\?\^_`{\|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+$/i;`

Reply:
I checked isEmail function, its allow some special characters, e.g: !, #, %, &, ...

https://github.com/validatorjs/validator.js/blob/8c4b3b35c3bd36dc3983aadb045422847154451b/src/lib/isEmail.js#L19-L25

Reply:
I've checked isEmail function, its allow some special characters, e.g: !, #, $, %, &, ...
![image](https://user-images.githubusercontent.com/88080925/127355267-1fb4e272-f299-4608-9f67-75db3ff39598.png)

Reply:
There are [certain special characters](https://www.w3resource.com/javascript/form/email-validation.php) that should be allowed in emails. For example, we use the plus sign `+` when we want to create additional accounts using the same email ex. `brandon+testing@bitholla.com`. I think at this time, we should use the `isEmail` function from `validator` library. The library still validates emails with rare invalid characters (`£`, `¶`, `§`, etc.) and there  is a [better library for email validation](https://www.npmjs.com/package/email-validator) but we would need to install it on the Kit repo.